### PR TITLE
Feat: Chart build history and status update

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

- API for Deployment history for devtron chart (installed in full mode in gitops way)
- On chart deploy and re-deploy (updates) new build history will be created,
- On chart default and bulk deploy will also create new build history in db.
- Status handler will update each build by git-hash.

Fixes # (issue): https://github.com/devtron-labs/devtron/issues/1297

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Chart Create
- [x] Chart Update
- [x] Chart Upgrade
- [x] Build History Fetch Api
- [x] Status Update Handler - Update Status For Each Build

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [x] I have tested it for all user roles
* [ ] I have added all requird unit/api test cases

### CODE REVIEW

File: api/appStore/AppStoreRouter.go [hunk 1]
```diff
@@ -83,4 +83,9 @@ func (router AppStoreRouterImpl) Init(configRouter *mux.Router) {
 
 	configRouter.Path("/cluster-component/install/{clusterId}").
 		HandlerFunc(router.deployRestHandler.DefaultComponentInstallation).Methods("POST")
+
+	configRouter.Path("/installed-app/deployment-history").Queries("installedAppId", "{installedAppId}").
```
Comment (pghildiyal) [inline api/appStore/AppStoreRouter.go:87]: it should be of the form `installed-app/${installedAppId}/deployment-history`

File: api/appStore/AppStoreDeploymentRestHandler.go [hunk 2]
```diff
@@ -467,3 +469,39 @@ func (handler *InstalledAppRestHandlerImpl) FetchAppDetailsForInstalledApp(w htt
...
+	response, err := handler.installedAppService.GetInstalledAppVersionHistory(installedAppId)
+	if err != nil {
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+
+	common.WriteJsonResp(w, err, response, http.StatusOK)
+}
+
+func (handler *InstalledAppRestHandlerImpl) GetDeploymentHistoryValues(w http.ResponseWriter, r *http.Request) {
```
Comment (pghildiyal) [inline api/appStore/AppStoreDeploymentRestHandler.go:491]: auth is missing

File: api/appStore/AppStoreDeploymentRestHandler.go [hunk 3]
```diff
@@ -467,3 +469,39 @@ func (handler *InstalledAppRestHandlerImpl) FetchAppDetailsForInstalledApp(w htt
 	}
 	common.WriteJsonResp(w, err, appDetail, http.StatusOK)
 }
+
+func (handler *InstalledAppRestHandlerImpl) GetDeploymentHistory(w http.ResponseWriter, r *http.Request) {
```
Comment (pghildiyal) [inline api/appStore/AppStoreDeploymentRestHandler.go:473]: auth is missing

File: pkg/appStore/AppStoreDeploymentService.go [hunk 4]
```diff
@@ -1098,3 +1098,86 @@ func (impl *InstalledAppServiceImpl) FindAppDetailsForAppstoreApplication(instal
...
+	return true, nil
+}
+
+func (impl InstalledAppServiceImpl) GetInstalledAppVersionHistoryValues(installedAppVersionHistoryId int) (*appStoreBean.IAVHistoryValues, error) {
+	versionHistory, err := impl.installedAppRepositoryHistory.GetInstalledAppVersionHistory(installedAppVersionHistoryId)
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("error while fetching installed version history", "error", err)
+		return nil, err
+	}
+	values := &appStoreBean.IAVHistoryValues{
```
[the remark this answers is not in the document]
Comment (vikramdevtron) [reply pkg/appStore/AppStoreDeploymentService.go:1195]: fixed

File: api/appStore/AppStoreDeploymentRestHandler.go (hunk 3 above)
Comment (vikramdevtron) [reply api/appStore/AppStoreDeploymentRestHandler.go:473]: done, added now

File: api/appStore/AppStoreDeploymentRestHandler.go (hunk 2 above)
Comment (vikramdevtron) [reply api/appStore/AppStoreDeploymentRestHandler.go:504]: done, added now

File: api/appStore/AppStoreRouter.go (hunk 1 above)
Comment (vikramdevtron) [reply api/appStore/AppStoreRouter.go:87]: will have separate exercise where we refactor app store related urls

**Review by pghildiyal — APPROVED** [review]

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**